VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -1319,6 +1319,514 @@ Function Install-EsxiCertificate {
+                '
+                $resourcesBodyObject += [pscustomobject]@{
+                    resources = $resourcesObject
+                }
+                $resourcesBodyObject | ConvertTo-Json -Depth 10 | Out-File -FilePath $tempPath"temp.json"
+                Get-Content $tempPath"temp.json" | Select-Object -Skip 1 | Set-Content $tempPath"temp1.json"
+                $resouresJson = Get-Content $tempPath"temp1.json" -Raw
+                $requestCertificateSpecJson = $caTypeJson + $resouresJson
+                $requestCertificateSpecJson | Out-File $tempPath"$($workloadDomain)-requestCertificateSpec.json"
+
+                Write-Output "Requesting Certificates for components associated with Domain $($workloadDomain)."
+                $myTask = Request-VCFCertificate -domainName $($workloadDomain) -json $tempPath"$($workloadDomain)-requestCertificateSpec.json"
+                Do {
+                    Write-Output "Checking status for the generation of signed certificates for components associated with workload domain ($($workloadDomain))..."
+                    Start-Sleep 6
+                    $response = Get-VCFTask $myTask.id
+                } While ($response.status -eq "IN_PROGRESS")
+                if ($response.status -eq "FAILED") {
+                    Write-Error "Workflow completed with status: $($response.status)." 
+                } elseif ($response.status -eq "SUCCESSFUL") {
+                    Write-Output "Workflow completed with status: $($response.status)."
+                } else {
+                    Write-Warning "Workflow completed with an unrecognized status: $($response.status). Please check the state before proceeding."
+                }
+                Write-Output "Request signed certficates for the components associated with workload domain $($workloadDomain) completed with status: $($response.status)."
+
+                # Remove the temporary directory.
+			    Remove-Item -Recurse -Force $tempPath  | Out-NULL
+        } else {
+            Write-Error "Unable to authenticate to SDDC Manager ($($server)): PRE_VALIDATION_FAILED."
+        }
+    } else {
+        Write-Error "Unable to connect to SDDC Manager ($($server)): PRE_VALIDATION_FAILED."
+    }
+}
+
+Function Install-SddcCertificate {
+    <#
+        .SYNOPSIS
+        Connect to SDDC Manager to install/replace the signed certificates for all components associated with the given workload domain.
+
+        .DESCRIPTION
+        The Install-SddcCertificate will request SDDC Manager to connect to Certificate Authority to sign the generated Certificate Signing Request files for all components associated with the given workload domain.
+
+        .EXAMPLE
+        Install-SddcCertificate -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -workloadDomain sfo-w01
+        This example will connect to Sddc Manager to install/replace the signed certificates for a given workload domain.
```

Comment:
```suggestion
        This example will connect to SDDC Manager to install/replace the signed certificates for a given workload domain.
```

Reply:
addressed in (c468fc14c2f684881d0da7753d1082bc7e9e70fc)